File: 73-set-matrix-zeroes/73-set-matrix-zeroes.rs
pub fn reset_to_zero(matrix: &mut Vec<Vec<i32>>, x: usize, y: usize){
    let m = matrix.len();
    let n = matrix[0].len();
    
    for i in 0..m{
        matrix[i][y] = 0;
    }
    
    for j in 0..n{
        matrix[x][j] = 0;
    }
}

impl Solution {
    pub fn set_zeroes(matrix: &mut Vec<Vec<i32>>) {
        let mut zeroes = vec![];
        
        let m = matrix.len();
        let n = matrix[0].len();
        
        for i in 0..m{
            for j in 0..n{
                if matrix[i][j] == 0{
                    zeroes.push((i, j));
                }
            }
        }
        
        // println!("{:?}", zeroes);
        
        for (x, y) in zeroes.iter(){
            reset_to_zero(matrix, *x, *y);
        }
        
    }
}

File: 73-set-matrix-zeroes/73-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        m, n = len(matrix), len(matrix[0])
        
        rows = set()
        cols = set()
        
        for i in range(0, m):
            for j in range(0, n):
                if matrix[i][j] == 0:
                    rows.add(i)
                    cols.add(j)
                    
        for i in rows:
            for j in range(0, n):
                matrix[i][j] = 0
                
        for j in cols:
            for i in range(0, m):
                matrix[i][j] = 0